---
title: "Circleries"
author: "Vincent Somerville"
date: "02/10/2020"
output: html_document
---

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)
```


## Circleries steps
1. check (bacterial) contig
  - blast for plasmid
    - OriC
  - blast/diamond bacterial housekeeping genes
    - 16S
  - BUSCO (maybe very generic...which contig is what)
  - contig size
  - check for presence of dnaA (diammond/PROKKA)
  ==> Take home: contig originates from?
2. check if potentially circler
  - map ends (10kb) onto eachother (minimap2) --> overlap or not
  - merge contig ends and map raw reads+ look for:
    - even coverage (output: X coverage--> is this in the range)
    - spanning reads (number of reads)
      - PE (are pairs mapping)
      - ONT/Pacbio (is read spanning)
    ==> Take home: contig is circler or not! 
3. circleries
  - take dnaA diamond mapping (+prodigal)
  - check if no genes are on reverse strand (prodigal)
  - check for presence of DNAa 
  - check for DNAa orientation
  - cut and shift fasta start
4. Polish the newly stitched region
  - QC how many locations have change!
  - advise if additional polishing is required!

## Dependencies

```{bash}


#conda activate circleries


###---------------------------------
## Part 1. 
###---------------------------------

busco

###---------------------------------
## Part 2. 
###---------------------------------

minimap2 --version


###---------------------------------
## Part 3. 
###---------------------------------

fasta_shift --help
prodigal --version

###---------------------------------
## Part 4. 
###---------------------------------

#PROKKA --version
#bwa --version


```

## parameter

Here, I start defining necessary bash variables.
```{bash}
circleriesPATH=/home/vincent/Desktop/Projects/2020_circleries
genomeFASTAname=03_test_Data/test_streptococcus_polished_dnaA-pos-1526474_nonOVERLAPING.fasta
outputFolderName=04_OUTPUTTest
```




## 1. check if bacterial contig

#### 1.1 contig description

Here, I quickly look for the number, name and length of contigs in the file. 
Therefore I worked on a nice onliner from [Daniel Cook](https://www.danielecook.com/generate-fasta-sequence-lengths/)

```{bash}

##================================
##header
##================================

cd ${circleriesPATH}

##================================
##create output folders
##================================

mkdir -p ${outputFolderName}
mkdir -p ${outputFolderName}/genomes
##================================
##contig length
##================================

cat ${genomeFASTAname} | awk '$0 ~ ">" {if (NR > 1) {print c;} c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' > ${outputFolderName}/genome.log
cat ${outputFolderName}/genome.log

##================================
##change fasta sequence onto one line
##this is useful in order to easly and quickly extract one contig after enother
##here we also change the fasta header to contain no whitespaces or tabs
##================================


 sed '/>/s/\t/_/g' 03_test_Data/${genomeFASTAname}|sed '/>/s/ /_/g'|sed '/>/s/$/\t/g' | tr -d '\n' | sed 's/\t/\n/g' > ${outputFolderName}/genomes/tmp_wide_all.fasta

##================================
##split every contig into one row
##================================

for header in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta)
do

header_short=$(echo ${header}|sed 's/>//g' )
grep "${header}" -A 1 ${outputFolderName}/genomes/tmp_wide_all.fasta) > ${outputFolderName}/genomes/${header_short}.fasta)

done



```

  
#### 1.2 dnaA search

Here, I try to diammond dnaA
Therefore I first create a dnaA database. 

```{bash}

##================================
##create dnaA DB
##================================



##================================
##header
##================================

cd ${circleriesPATH}

##================================
##create output folders
##================================

mkdir -p ${outputFolderName}
mkdir -p ${outputFolderName}/genomes
##================================
##contig length
##================================

cat ${genomeFASTAname} | awk '$0 ~ ">" {if (NR > 1) {print c;} c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' > ${outputFolderName}/genome.log
cat ${outputFolderName}/genome.log

##================================
##change fasta sequence onto one line
##this is useful in order to easly and quickly extract one contig after enother
##here we also change the fasta header to contain no whitespaces or tabs
##================================


 sed '/>/s/\t/_/g' 03_test_Data/${genomeFASTAname}|sed '/>/s/ /_/g'|sed '/>/s/$/\t/g' | tr -d '\n' | sed 's/\t/\n/g' > ${outputFolderName}/genomes/tmp_wide_all.fasta

##================================
##split every contig into one row
##================================

for header in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta)
do

header_short=$(echo ${header}|sed 's/>//g' )
grep "${header}" -A 1 ${outputFolderName}/genomes/tmp_wide_all.fasta) > ${outputFolderName}/genomes/${header_short}.fasta)

done


```





  - blast for plasmid
    - OriC
  - blast/diamond bacterial housekeeping genes
    - 16S
 

```{bash}

```

 ==> Take home: contig originates from?

## 2. show circlarity

#### 2.1 map ends
```{bash}




```
#### 2.3 align reads

```{bash}



```


## 3. Circleries

use [fasta_shift](https://github.com/b-brankovics/fasta_tools).

```{bash}
##===========
##PROKKA annotation
##===========

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/
#mkdir -p /data/Project/2020_StarterCultureDiversity/06_PROKKA/${speciesss}/${culture}_${strainzzz}

/usr/bin/perl /usr/bin/prokka --cpus 5 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/ --locustag ${sample} --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta


##===========
##QC
##===========


seq_length.py /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*.fna
grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}'


##===========
##start align
##===========



 echo ${sample}
 
 grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}'
 
 
 orientation=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |cut -f 7)
 contigName=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |cut -f 1)
 samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${contigName} > \
        /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna
  rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned
  mkdir -p /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned
  if [ $orientation == "+" ]
    then              
      echo "forward oriented"
      startAlign=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |awk -F "\t" '{print $4-5}')
      echo -e "start align at:  "${startAlign}
      /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} > \
              /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta
  else              
      echo "reverse oriented"
      startAlign=$(grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/*gff |awk -F "\t" '{if($3=="gene")print $0}' |awk -F "\t" '{print $5+5}')
      echo -e "start align at:  "${startAlign}
      /home/vincent/apps/fasta_tools/bin/fasta_shift -i /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/bac_genome.fna -p ${startAlign} > \
      /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta
      revseq -sequence /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/tmp.fasta -outseq \
          /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta -notag
  fi  
  
###-------------------QC if DNA is really at first position
  

samtools faidx /data/Project/2020_StarterCultureDiversity/09_flye/Polishing_woCirclator/${sample}/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta ${non_bac} >> \
        /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta


###-------------------QC if DNA is really at first position

rm -r /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter n/perl /usr/bin/prokka --cpus 35 --outdir /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter --locustag ${sample} --proteins s.faa --evalue 0.001 --addgenes /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta



echo ${sample}
grep "dnaA" /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/ProkkaAfter/*gff |awk -F "\t" '{if($3=="gene")print $0}'

```

## 4 polish edge


## appendix


#### 1.x BUSCO 

Here, I implement the usage of BUSCO in order to see what the contigs are. 

I have been having many dependencie issues with busco. therefore I currently but it aside. 

```{bash}


##================================
##header
##================================

cd ${circleriesPATH}

##================================
##create output folders
##================================

mkdir -p ${outputFolderName}/busco
cd ${outputFolderName}/busco

##================================
##script
##================================

#busco -i [SEQUENCE_FILE] -l [LINEAGE] -o [OUTPUT_NAME] -m [MODE] [OTHER OPTIONS]
busco -i ${genomeFASTAname} -l [LINEAGE] -o [OUTPUT_NAME] -m genome [OTHER OPTIONS]

```


