---
title: "Circleries"
author: "Vincent Somerville"
date: "02/10/2020"
output: html_document
---

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)
```


## Circleries steps
1. check (bacterial) contig
  - blast for plasmid
    - OriC
  - blast/diamond bacterial housekeeping genes
    - 16S
  - BUSCO (maybe very generic...which contig is what)
  - contig size
  - check for presence of dnaA (diammond/PROKKA)
  ==> Take home: contig originates from?
2. check if potentially circler
  - map ends (10kb) onto eachother (minimap2) --> overlap or not
  - merge contig ends and map raw reads+ look for:
    - even coverage (output: X coverage--> is this in the range)
    - spanning reads (number of reads)
      - PE (are pairs mapping)
      - ONT/Pacbio (is read spanning)
    ==> Take home: contig is circler or not! 
3. circleries
  - take dnaA diamond mapping (+prodigal)
  - check if no genes are on reverse strand (prodigal)
  - check for presence of DNAa 
  - check for DNAa orientation
  - cut and shift fasta start
4. Polish the newly stitched region
  - QC how many locations have change!
  - advise if additional polishing is required!

## Dependencies

```{bash dependencies , eval=TRUE}


#conda activate circleries


###---------------------------------
## Part 1. 
###---------------------------------

#busco

###---------------------------------
## Part 2. 
###---------------------------------

minimap2 --version
#htsbox
#bwa

###---------------------------------
## Part 3. 
###---------------------------------

fasta_shift_path=/home/vincent/Desktop/Projects/2020_circleries/02_dependencies/fasta_shift
$fasta_shift_path --help

revseq #embos toolkit

#prodigal --version

###---------------------------------
## Part 4. 
###---------------------------------

#PROKKA --version
#bwa --version


```

## parameter

Here, I start defining necessary bash variables.
```{bash parameters, eval=TRUE}
circleriesPATH=/home/vincent/Desktop/Projects/2020_circleries
genomeFASTAname=03_test_Data/test_streptococcus_polished_dnaA-pos-1526474_nonOVERLAPING.fasta
flyeOutput=
outputFolderName=04_OUTPUTTest
threads=8
outName=Final_startAligned
```

## 1. check if bacterial contig

#### 1.1 contig description

Here, I quickly look for the number, name and length of contigs in the file. 
Therefore I worked on a nice oneliner from [Daniel Cook](https://www.danielecook.com/generate-fasta-sequence-lengths/)

```{bash}

##================================
##header
##================================

cd ${circleriesPATH}

##================================
##create output folders
##================================

mkdir -p ${outputFolderName}
mkdir -p ${outputFolderName}/genomes
##================================
##contig length
##================================

#cat ${genomeFASTAname} | awk '$0 ~ ">" {if (NR > 1) {print c;} c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' > ${outputFolderName}/genome.log
#cat ${outputFolderName}/genome.log

##================================
##change fasta sequence onto one line
##this is useful in order to easly and quickly extract one contig after enother
##here we also change the fasta header to contain no whitespaces or tabs
##================================

 sed '/>/s/\t/_/g' 03_test_Data/${genomeFASTAname}|sed '/>/s/ /_/g'|sed '/>/s/$/\t/g' | tr -d '\n' | sed 's/\t/\n/g' > ${outputFolderName}/genomes/tmp_wide_all.fasta

##================================
##split every contig into one row
##================================

for header in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta)
do

header_short=$(echo ${header}|sed 's/>//g' )
grep "${header}" -A 1 ${outputFolderName}/genomes/tmp_wide_all.fasta) > ${outputFolderName}/genomes/${header_short}.fasta)

done

##================================
##make prodigal 
##================================

#rm -r ${outputFolderName}/genomes/annotations/
#mkdir -p ${outputFolderName}/genomes/annotations/

#prodigal -i ${outputFolderName}/genomes/tmp_wide_all.fasta -o ${outputFolderName}/genomes/annotations/prodigal_out -a ${outputFolderName}/genomes/annotations/prodigal_out_my.proteins.faa

```

  
#### 1.2 dnaA search
##### 1.2.1 DB creation

Here, I try to diammond dnaA
Therefore I first create a dnaA database. 

```{bash}
##================================
##create dnaA DB
##================================

mkdir -p 99_OrthoDB
cd 99_OrthoDB
wget https://v101.orthodb.org/download/odb10v1_OGs.tab.gz
wget https://v101.orthodb.org/download/odb10v1_OG2genes.tab.gz
wget https://v101.orthodb.org/download/odb10v1_genes.tab.gz
gunzip *.gz

cd ${circleriesPATH}


mkdir -p 99_OMA
cd 99_OMA
wget https://omabrowser.org/All/oma-protein-annotations.txt.gz


wget https://omabrowser.org/All/group-descriptions.txt.gz
#wget https://omabrowser.org/All/All.cog.tar.gz



##---------1. OG description
wget https://omabrowser.org/All/group-descriptions.txt.gz
##---------2. similiar OG
wget https://omabrowser.org/All/oma-close-groups.txt.gz
##---------3. file that contains the seqNames for the OG
wget https://omabrowser.org/All/oma-groups.txt.gz
##---------4. the actual faa sequence file with the seqNames as header
wget https://omabrowser.org/All/oma-seqs.fa.gz


###----------------dnaA groups
grep "dnaa" -i  group-descriptions.txt |grep "regulatory"  -v |grep "domain-containing" -v|grep "DnaD" -v


###----------------check if more similar dnaA groups exist
grep "dnaa" -i  group-descriptions.txt |grep "regulatory"  -v |grep "domain-containing" -v|grep "DnaD" -v
grep "OMAGrp921125 " oma-close-groups.txt |cut -f 1

rm all_groups_very_similiar.txt
for groupsss in $(grep "dnaa" -i  group-descriptions.txt |grep "regulatory"  -v |grep "domain-containing" -v|grep "DnaD" -v|cut -f 1)
do
echo ${groupsss}
grep "OMAGrp${groupsss} " oma-close-groups.txt |cut -f 1 >> all_groups_very_similiar.txt

echo -e "OMAGrp"${groupsss} >> all_groups_very_similiar.txt

done

for groupsss in $(grep "dnaa" -i  group-descriptions.txt |grep "regulatory"  -v |grep "domain-containing" -v|grep "DnaD" -v|cut -f 1)
do
echo ${groupsss}
grep "OMAGrp0${groupsss} " oma-close-groups.txt |cut -f 1 >> all_groups_very_similiar.txt

echo -e "OMAGrp0"${groupsss} >> all_groups_very_similiar.txt

done

grep "dnaa" -i  group-descriptions.txt |grep "regulatory"  -v |grep "domain-containing" -v|grep "DnaD" -v

sort all_groups_very_similiar.txt|sed 's/://g'|uniq  
sort all_groups_very_similiar.txt|sed 's/://g'|uniq |sed 's/OMAGrp//g' |sed 's/^0//g'|sort -n|uniq

##-------------------------------------check with extensive OG list what they are


for OGSss in $(sort all_groups_very_similiar.txt|sed 's/://g'|uniq |sed 's/OMAGrp//g' |sed 's/^0//g'|sort -n|uniq )
do

awk -v OGzzz="${OGSss}" '{if($1==OGzzz)print $0}'  group-descriptions.txt

done
##many of these genes are strange I will therefore not take them and only use the original list




##-------------------------------------take the OG labels and loop through the oma groups description to extract the seqNames to download

rm list_of_seq_names_dnaA.txt
for OGSss in $(grep "dnaa" -i  group-descriptions.txt |grep "regulatory"  -v |grep "domain-containing" -v|grep "DnaD" -v|cut -f 1)
do

echo ${OGSss}
awk -v OGzzz="${OGSss}" '{if($1==OGzzz) print $0}' oma-groups.txt | awk '{$1=$2=""; print $0}'  | sed 's/^ //g'| sed 's/^ //g'|sed 's/ /\n/g' >> list_of_seq_names_dnaA.txt


done

##-------------------------------------extract sequences with SeqNames


 sed '/>/s/$/\t/g' oma-seqs.fa | tr -d '\n' | sed 's/\t/\n/g' > oma-seqs_online.fa
rm oma-seqs.fa
sed -i '2,$s/>/\n>/' oma-seqs_online.fa
rm oma-seqs_online_dnaA.fa
for SequNamesss in $(cat list_of_seq_names_dnaA.txt |sort|uniq)
do

echo ${SequNamesss}
grep "${SequNamesss}" -A 1 oma-seqs_online.fa >>  oma-seqs_online_dnaA.fa

done



sed 's/> />/g' 99_OMA/oma-seqs_online_dnaA.fa  > 99_OMA/oma-seqs_online_dnaA_corrected.fa

makeblastdb -max_file_sz 1GB -in 99_OMA/oma-seqs_online_dnaA_corrected.fa \
 -out 99_OMA/oma-seqs_online_dnaA_corrected -parse_seqids -dbtype prot




###don't print the first or second column 
grep oma-groups.txt.gz | awk '{$1=$2=""; print $0}' 


gunzip *gz

grep "dnaa" -i group-descriptions.txt 


###head
head -5 oma-protein-annotations.txt
head -5 oma-close-groups.txt
head -5 group-descriptions.txt


group-descriptions.txt

grep "dnaa" -i  oma-protein-annotations.txt

grep "30176" -i  oma-close-groups.txt
grep "30176" -i  oma-protein-annotations.txt

seqlength.py 99_OMA/oma-seqs_online_dnaA_corrected.fa

##================================
##create dnaA DB from OMA
##================================

#1. download a MSA from a dnaA homology group 

mkdir -p 99_OMA

###------------------------------------
#make MSA
###------------------------------------

mafft --thread 10 --maxiterate 1000  --auto  99_OMA/20210206_dnaA_OMA.faa >  99_OMA/20210206_dnaA_OMA_alg.faa


###------------------------------------
#create HMM database
###------------------------------------

hmmbuild  99_OMA/20210206_dnaA_OMA_alg.hmm  99_OMA/20210206_dnaA_OMA_alg.faa

##================================
##download trycycler DNA start site
##================================
mkdir -p 05_startAlign_data
cd 05_startAlign_data

wget https://github.com/rrwick/Trycycler/raw/main/trycycler/data/starting_genes.fasta

```

##### 1.2.2 sequence preperation

Here, I quickly look for the number, name and length of contigs in the file. 
Therefore I worked on a nice oneliner from [Daniel Cook](https://www.danielecook.com/generate-fasta-sequence-lengths/)

Further I check for the presence of dnaA. 
I also run a Busco to identify the bacterial contigs. 

```{bash prepperation , eval=FALSE}
##================================
##header
##================================

cd ${circleriesPATH}

##================================
##create output folders
##================================

mkdir -p ${outputFolderName}
mkdir -p ${outputFolderName}/genomes
##================================
##contig length
##================================

cat ${genomeFASTAname} | awk '$0 ~ ">" {if (NR > 1) {print c;} c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' > ${outputFolderName}/genome.log
cat ${outputFolderName}/genome.log

##================================
##change fasta sequence onto one line
##this is useful in order to easly and quickly extract one contig after enother
##here we also change the fasta header to contain no whitespaces or tabs
##================================

 sed '/>/s/\t/_/g' ${genomeFASTAname}|sed '/>/s/ /_/g'|sed '/>/s/$/\t/g' | tr -d '\n' | sed 's/\t/\n/g' > ${outputFolderName}/genomes/tmp_wide_all.fasta

##================================
##split every contig into one row
##================================

for header in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta)
do

echo $header
mkdir -p  ${outputFolderName}/genomes/contigs_split/
header_short=$(echo ${header}|sed 's/>//g' )
grep "${header}" -A 1 ${outputFolderName}/genomes/tmp_wide_all.fasta > ${outputFolderName}/genomes/contigs_split/${header_short}.fasta

done
```

##### 1.2.3 dnaA Search

```{bash MINIMAP2 , eval=FALSE}
##================================
##minimap2 ORIs to sequences
##================================


for header in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta)
do

echo $header

#bwa index ${outputFolderName}/genomes/contigs_split/${header_short}.fasta
#bwa index ${circleriesPATH}/05_startAlign_data/starting_genes.fasta
mkdir -p ${outputFolderName}/start_alignment_mapping/
  
##map dnaA genes to the indiviual references and remove non-mapping or low quality mapping reads  
#bwa mem -x intractg ${outputFolderName}/genomes/contigs_split/${header_short}.fasta ${circleriesPATH}/05_startAlign_data/starting_genes.fasta |samtools view -q 30 -F 4 > ${outputFolderName}/start_alignment_mapping/${header_short}.mapping


##map dnaA genes to the individual references and remove non-mapping or low quality mapping reads (mapq>50) 
minimap2 ${outputFolderName}/genomes/contigs_split/${header_short}.fasta ${circleriesPATH}/05_startAlign_data/starting_genes.fasta  |awk -F "\t" '{OFS="\t"}{if($12>50) print $0}'  > ${outputFolderName}/start_alignment_mapping/${header_short}.minimap



done

##================================
##minimap2 analysis
##================================

echo -e "#contigName\tContigOrigin\tNumberOfdnaAmappings\tOrientationOfMapping\tStartOfTargetMapping\tContigLength" >  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap

for header in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta|sed 's/>//g')
do

echo $header


#the longest alignment shows the orientation of the contig
#sort -k10 -n -r ${outputFolderName}/start_alignment_mapping/${header_short}.minimap

#find out if there are mappings
number_of_mappings=$(wc -l ${outputFolderName}/start_alignment_mapping/${header_short}.minimap|cut -d ' ' -f 1)
orientation=$(sort -k10 -n -r ${outputFolderName}/start_alignment_mapping/${header_short}.minimap|head -1 |cut -f 5)


  
if [ $number_of_mappings == "0" ]
then
        echo "no dnaA map found"
      
       echo -e ${header}"\tnon-bacterial_contig\t"${number_of_mappings}"\tNA\tNA\tNA" >>  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap
    
elif [ $orientation == "+" ]
then

  #the most left alignment this shows the start of the gene
  #sort -k8 -n ${outputFolderName}/start_alignment_mapping/${header_short}.minimap

  start=$(sort -k8 -n ${outputFolderName}/start_alignment_mapping/${header_short}.minimap|head -1 |awk -F "\t" '{OFS="\t"}{print $8-$3-5}')
        contig_length=$(sort -k8 -n ${outputFolderName}/start_alignment_mapping/${header_short}.minimap|head -1 |cut -f 7)
        
        echo -e ${header}"\tbacterial_contig\t"${number_of_mappings}"\t"${orientation}"\t"${start}"\t"${contig_length} >>  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap
        
else

    #the most right alignment this shows the start of the gene
    #sort -k9 -n ${outputFolderName}/start_alignment_mapping/${header_short}.minimap

    start=$(sort -k9 -n -r ${outputFolderName}/start_alignment_mapping/${header_short}.minimap|head -1 |awk -F "\t" '{OFS="\t"}{print $9+$2-$4+5}')
        contig_length=$(sort -k9 -n -r  ${outputFolderName}/start_alignment_mapping/${header_short}.minimap|head -1 |cut -f 7)
        
        echo -e ${header}"\tbacterial_contig\t"${number_of_mappings}"\t"${orientation}"\t"${start}"\t"${contig_length} >>  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap
        
fi


done

cat ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap
cut -f 3  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap

```


```{bash OLDmAPPING ,eval=FALSE}
##================================
##blastn with DB
##================================

#sed 's/> />/g' 99_OMA/oma-seqs_online_dnaA.fa  > 99_OMA/oma-seqs_online_dnaA_corrected.fa

#makeblastdb -max_file_sz 1GB -in 99_OMA/oma-seqs_online_dnaA_corrected.fa \
 -out 99_OMA/oma-seqs_online_dnaA_corrected -parse_seqids -dbtype prot


#mkdir -p ${outputFolderName}/blast/

# blastx  -max_hsps 1 -max_target_seqs 1 -show_gis \
  -query ${outputFolderName}/genomes/tmp_wide_all.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc" \
  -db ${circleriesPATH}/99_OMA/oma-seqs_online_dnaA_corrected \
  -out ${outputFolderName}/blast/dnaA_matches.blast \
  -evalue 0.01 -word_size 3


##================================
##Hmm with dnaA DB
##================================
#rm -r ${outputFolderName}/genomes/hmm_dnaA
#mkdir -p ${outputFolderName}/genomes/hmm_dnaA

#hmmsearch  -o ${outputFolderName}/genomes/hmm_dnaA/dna_idnentificaion.txt   99_OMA/20210206_dnaA_OMA_alg.hmm ${outputFolderName}/genomes/annotations/prodigal_out_my.proteins.faa


```

##### 1.2.4 search and identify plasmids/virsuses

  - blast for plasmid
    - OriC
  - blast/diamond bacterial housekeeping genes
    - 16S
 

```{bash non-bacterial-contigs-start ,eval=FALSE}

##not started yet
```


##### 1.2.5 summary contigs

 ==> Take home: contig originates from?
 With the circular contigs I would go into the circluarizse step. 

```{bash}
##===========================================
#summary stats before circularizing
##===========================================

echo "============================================================"
echo "summary stats before circularizing"
echo "============================================================"

echo -e "We will proceed start aligning following " ${BacContigs} " contigss..."
grep "^#"  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap 
grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2=="bacterial_contig")print $0}'


```



## 2. show circlarity

Not incorporated yet. 

#### 2.1 map ends
```{bash}




```
#### 2.3 align reads

```{bash}



```


## 3. Circleries


#### 3.1 Circleries contigs

use [fasta_shift](https://github.com/b-brankovics/fasta_tools).

```{bash Circleries, eval=FALSE}
##===========
##start align
##===========

 rm -r ${outputFolderName}/StartAlignedContigs/
 mkdir -p ${outputFolderName}/StartAlignedContigs/
 mkdir -p ${outputFolderName}/tmp/
  
for contigName in $(grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2=="bacterial_contig")print $1}')
do

echo -e "contig name: \t\t" ${contigName}
 orientation=$(awk -F "\t" -v contigzz="$contigName" '{OFS="\t"}{if($1==contigzz)print $4}' ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap )
 contigStart=$(awk -F "\t" -v contigzz="$contigName" '{OFS="\t"}{if($1==contigzz)print $5}'   ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap )
 contigLength=$(awk -F "\t" -v contigzz="$contigName" '{OFS="\t"}{if($1==contigzz)print $6}'  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap )

  if [ $orientation == "+" ]
    then              
      echo "forward oriented and start align at:  "${contigStart}

      $fasta_shift_path -i ${outputFolderName}/genomes/contigs_split/${contigName}.fasta -p ${contigStart} > \
              ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta
  else              

        echo "reverse oriented and start align at:  "${contigStart}

      $fasta_shift_path -i ${outputFolderName}/genomes/contigs_split/${contigName}.fasta -p ${contigStart} > \
              ${outputFolderName}/tmp/${contigName}_startAligned_wrongOrientation.fasta
      
      revseq -sequence ${outputFolderName}/tmp/${contigName}_startAligned_wrongOrientation.fasta -outseq \
          ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta -notag
      
      #add startaligned tag

    #  grep ">" ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta
      sed -i '/^>/ s/$/_StartAligned/' ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta 
    #  grep ">" ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta


  fi  



done
```

#### 3.2 QC of Circleries 


```{bash QC_Circleries, eval=FALSE}
###-------------------QC if DNA is really at first position
  
echo -e "#contigName\tContigOrigin\tNumberOfdnaAmappings\tOrientationOfMapping\tStartOfTargetMapping\tContigLength" >  ${outputFolderName}/start_alignment_mapping/After_StartAlignment_contigs.minimap

for contigName in $(grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2=="bacterial_contig")print $1}')
do

minimap2 ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta  ${circleriesPATH}/05_startAlign_data/starting_genes.fasta |awk -F "\t" '{OFS="\t"}{if($12>50) print $0}'  > ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap

sort -k10 -n -r ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap |head -1


number_of_mappings=$(wc -l ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap|cut -d ' ' -f 1)
orientation=$(sort -k10 -n -r ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap|head -1 |cut -f 5)

  
if [ $number_of_mappings == "0" ]
then
        echo "no dnaA map found"
      
       echo -e ${header}"\tnon-bacterial_contig\t"${number_of_mappings}"\tNA\tNA\tNA" >>  ${outputFolderName}/start_alignment_mapping/After_StartAlignment_contigs.minimap
    
elif [ $orientation == "+" ]
then
           echo "Looks good!"

  start=$(sort -k8 -n ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap|head -1 |awk -F "\t" '{OFS="\t"}{print $8-$3-5}')
        contig_length=$(sort -k8 -n ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap|head -1 |cut -f 7)
        
        echo -e ${header}"\tbacterial_contig\t"${number_of_mappings}"\t"${orientation}"\t"${start}"\t"${contig_length} >>  ${outputFolderName}/start_alignment_mapping/After_StartAlignment_contigs.minimap
        
else

    start=$(sort -k9 -n -r ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap|head -1 |awk -F "\t" '{OFS="\t"}{print $9+$2-$4+5}')
        contig_length=$(sort -k9 -n -r  ${outputFolderName}/start_alignment_mapping/${header_short}_afterStartAlignment.minimap|head -1 |cut -f 7)
        
        echo -e ${header}"\tbacterial_contig\t"${number_of_mappings}"\t"${orientation}"\t"${start}"\t"${contig_length} >>  ${outputFolderName}/start_alignment_mapping/After_StartAlignment_contigs.minimap
        
        echo "SOMETHING IS WRONG BECAUSE dnaA IS STILL ON THE REVERSE STRAND"
fi

done

```


#### 3.2 bring contigs together again

Here, I finally merge the contigs again. 

```{bash merge_Circleries, eval=FALSE}

rm  ${outputFolderName}/${outName}.fasta
for contigName in $(grep ">" ${outputFolderName}/genomes/tmp_wide_all.fasta|sed 's/>//g')
do

origin=$(awk -F "\t" -v contigzz="$contigName" '{OFS="\t"}{if($1==contigzz)print $2} ' ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap)


 if [ $origin=="bacterial_contig" ]
    then              
     echo -e "Bacterial contig name: \t\t" ${contigName}
     cat ${outputFolderName}/StartAlignedContigs/${contigName}_startAligned.fasta  >> ${outputFolderName}/${outName}.fasta

  else              

     echo -e "Non-bacterial contig name: \t\t" ${contigName}
     cat  ${outputFolderName}/genomes/${contigName}.fasta  >> ${outputFolderName}/${outName}.fasta


  fi  

done

#grep ">" ${outputFolderName}/${outName}.fasta

```


#### 3.3 Final_Info

Here, I finally merge the contigs again. 

```{bash finalInfo_Circleries, eval=FALSE}

echo "============================================================"
echo "Final Infos"
echo "============================================================"
here=$(pwd)
echo -e "Location of Startaligned contigs: \t\t"${here}/${outputFolderName}/${outName}".fasta"


numberContigs=$(grep "^#" -c ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap)
BacContigs=$(grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2=="bacterial_contig")print $0}'|wc -l)
NonBacContigs=$(grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2!="bacterial_contig")print $0}' |wc -l)

echo -e "Total number of contigs: \t\t"${numberContigs}
echo -e "Bacterial contigs: \t\t"${BacContigs}
#grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2=="bacterial_contig")print $0}'
echo -e "Non-Bacterial contigs: \t\t"${NonBacContigs}
#grep "^#" -v  ${outputFolderName}/start_alignment_mapping/StartAlignment_contigs.minimap |awk -F "\t" '{OFS="\t"}{if($2!="bacterial_contig")print $0}'


echo -e "The location of dnaA on the " ${BacContigs} " Bacterial contigs..."
grep "^#"  ${outputFolderName}/start_alignment_mapping/After_StartAlignment_contigs.minimap
grep "^#" -v  ${outputFolderName}/start_alignment_mapping/After_StartAlignment_contigs.minimap|awk -F "\t" '{OFS="\t"}{if($2=="bacterial_contig")print $0}'


echo -e "Remember the contigs should still be polished now..."

```


## 4 polish edge


## appendix


#### 1.x BUSCO 

Here, I implement the usage of BUSCO in order to see what the contigs are. 

I have been having many dependencie issues with busco. therefore I currently but it aside. 

```{bash}


##================================
##header
##================================

cd ${circleriesPATH}

##================================
##create output folders
##================================

mkdir -p ${outputFolderName}/busco
cd ${outputFolderName}/busco

##================================
##script
##================================

#busco -i [SEQUENCE_FILE] -l [LINEAGE] -o [OUTPUT_NAME] -m [MODE] [OTHER OPTIONS]
busco -i ${genomeFASTAname} -l [LINEAGE] -o [OUTPUT_NAME] -m genome [OTHER OPTIONS]

```


