---
title: "20211229_circles_analysis"
author: "Vincent Somerville"
date: "30/12/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)
```

## dnaA database

```{bash}
grep ">" -c /home/vincent/Desktop/Projects/2020_circleries/05_startAlign_data/starting_genes.fasta

grep "dnaa" -i -c  /home/vincent/Desktop/Projects/2020_circleries/05_startAlign_data/starting_genes.fasta
grep "chromo" -i -c  /home/vincent/Desktop/Projects/2020_circleries/05_startAlign_data/starting_genes.fasta

grep "plasmid" -i -c  /home/vincent/Desktop/Projects/2020_circleries/05_startAlign_data/starting_genes.fasta

grep ">" /home/vincent/Desktop/Projects/2020_circleries/05_startAlign_data/starting_genes.fasta

```


## check start alignment

Here, I check how many completely assembled genomes are start aligned on NCBI. 

```{bash}
mkdir -p /data/Project/2021_circlator/NCBI_analysis/log/
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt


cut -f 12 /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt|sort|uniq -c


##------------------------------------------------
#only chromosome
##------------------------------------------------

awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Chromosome" ) print $0}' /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 20 | sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome
    
    
cd 
rm -r /data/Project/2021_circlator/NCBI_analysis/download/gff/
 mkdir -p /data/Project/2021_circlator/NCBI_analysis/download/gff/
 
 rm -r /data/Project/2021_circlator/NCBI_analysis/analysis
 mkdir -p /data/Project/2021_circlator/NCBI_analysis/analysis/
 
 
cd /data/Project/2021_circlator/NCBI_analysis/download/gff/

#wget -i /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome


for downloadsss in $(cat /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome)
do

echo $downloadsss

namess=$(echo $downloadsss |sed 's/.*\///g'|sed 's/_genomic.gff.gz//g')

wget  ${downloadsss}


zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $0}' >> /data/Project/2021_circlator/NCBI_analysis/analysis/genes.identified.chromosome
startss=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $4}')
orientation=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $7}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq" && $4 < "5000" && $7 =="+")print "Yes"; else print "NO"}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq") && ($4 < "5000") && ( $7 =="+")) print "Yes" }')

startAligned=$(echo $startss | awk -v orientations="$orientation" '{if(($1 < 5000) && (orientations == "+")) print "Yes"; else print "NO" }')



#echo -e ${namess}
#echo -e ${namess}"\t"${startss}
#echo -e ${namess}"\t"${startss}"\t"${orientation}
#echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} 

#zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq" && $4 < "5000" && $7 =="+")) print "Yes" }'

#|awk -F "\t" -v contigNames="$contigName" '{OFS="\t"}{if($2=="bacterial_contig"  && ($7=="Y"||$8=="Y"||$9=="Y"))print "Y"; else print "N"}'

echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} >> /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome
#rm ${namess}*
done


cat /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome

cut -f 4 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome |sort|uniq -c
cut -f 3 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome |sort|uniq -c

wc -l /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome
wc -l /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome



##------------------------------------------------
#only complete genome
##------------------------------------------------

awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ) print $0}' /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 20 | sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_Complete_Genome
    
    
cd 
#rm -r /data/Project/2021_circlator/NCBI_analysis/download/gff/
#mkdir -p /data/Project/2021_circlator/NCBI_analysis/download/gff/
 
# rm -r /data/Project/2021_circlator/NCBI_analysis/analysis
# mkdir -p /data/Project/2021_circlator/NCBI_analysis/analysis/
 
 
cd /data/Project/2021_circlator/NCBI_analysis/download/gff/

#wget -i /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome


for downloadsss in $(cat /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_Complete_Genome)
do

echo $downloadsss

namess=$(echo $downloadsss |sed 's/.*\///g'|sed 's/_genomic.gff.gz//g')

wget  ${downloadsss}

zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $0}' >> /data/Project/2021_circlator/NCBI_analysis/analysis/genes.identified.Complete_Genome
startss=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $4}')
orientation=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $7}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq" && $4 < "5000" && $7 =="+")print "Yes"; else print "NO"}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq") && ($4 < "5000") && ( $7 =="+")) print "Yes" }')

startAligned=$(echo $startss | awk -v orientations="$orientation" '{if(($1 < 5000) && (orientations == "+")) print "Yes"; else print "NO" }')



#echo -e ${namess}
#echo -e ${namess}"\t"${startss}
#echo -e ${namess}"\t"${startss}"\t"${orientation}
#echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} 

#zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq" && $4 < "5000" && $7 =="+")) print "Yes" }'

#|awk -F "\t" -v contigNames="$contigName" '{OFS="\t"}{if($2=="bacterial_contig"  && ($7=="Y"||$8=="Y"||$9=="Y"))print "Y"; else print "N"}'

echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} >> /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome
#rm ${namess}*
done


cat /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome

cut -f 4 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome |sort|uniq -c
cut -f 3 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome |sort|uniq -c

wc -l /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome
wc -l /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_Complete_Genome


##----------------------------------------------------
#move local

mkdir -p  /home/vincent/Desktop/Projects/2020_circleries/07_figures
mkdir -p  /home/vincent/Desktop/Projects/2020_circleries/06_analysis/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome /home/vincent/Desktop/Projects/2020_circleries/06_analysis/

mkdir -p  /home/vincent/Desktop/Projects/2020_circleries/06_analysis/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome /home/vincent/Desktop/Projects/2020_circleries/06_analysis/


```


```{r}


library(tidyverse)

#-----------------------------------------------------------------------
#chromosome
#-----------------------------------------------------------------------

dnaA_genes_stats <- read_delim("~/Desktop/Projects/2020_circleries/06_analysis/dnaA_genes.stats.chromosome", delim = "\t", escape_double = FALSE, col_names = c("genome","startpostion","orientation","startAligned"), trim_ws = TRUE)

plot_chromsome <- ggplot(dnaA_genes_stats,aes(x=startAligned,fill=startAligned,color=startAligned))+geom_bar()+theme_classic()+labs(x="Genome start aligned\n at dnaA",y="Genomes",title = "Chromosome-level assemblies\non NCBI")+theme(legend.position = "none")

png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_chromosome_startAligned_NCBI.png", width = 2200, height = 3000,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plot_chromsome

dev.off()


#-----------------------------------------------------------------------
#complete genomes
#-----------------------------------------------------------------------

dnaA_genes_stats_complete <- read_delim("~/Desktop/Projects/2020_circleries/06_analysis/dnaA_genes.stats.Complete_Genome", delim = "\t", escape_double = FALSE, col_names = c("genome","startpostion","orientation","startAligned"), trim_ws = TRUE)

plot_complete_genome <- ggplot(dnaA_genes_stats_complete,aes(x=startAligned,fill=startAligned,color=startAligned))+geom_bar()+theme_classic()+labs(x="Genome start aligned\n at dnaA",y="Genomes",title = "Complete genome assemblies\non NCBI")+theme(legend.position = "none")
plot_complete_genome
png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_completeGenomes_startAligned_NCBI.png", width = 2200, height = 3000,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plot_complete_genome

dev.off()


#-----------------------------------------------------------------------
#together
#-----------------------------------------------------------------------

library(patchwork)
plot_complete_genome+plot_chromsome

png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_both_startAligned_NCBI.png", width = 4400, height = 3000,res=400)

plot_complete_genome+plot_chromsome

dev.off()

```


# biosample2SRA

download SRA files to test circlator and circleries on 20 long read data. 

from https://www.biostars.org/p/448799/ 

```{bash}
##--------------------------------------------------------
#test
##--------------------------------------------------------
cd /data/Project/2021_circlator/NCBI_analysis/log/
cat samples.txt 
SAMN14390563
SAMN14390566
SAMN14390576
SAMN14390578
SAMN14453547
SAMN14453553


epost -db biosample -input samples.test -format acc | \
elink -target sra | \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > samples.test.sra


/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout SRR390728 

/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout SRR390728 

SAMN14390563

```



# move circleries to bigmachine

```{bash}




scp -r /home/vincent/Desktop/Projects/2020_circleries vincent@130.223.51.66:/data/Project/2021_circlator/20220106_2020_circleries_tool/

cd /data/Project/2021_circlator/20220106_2020_circleries_tool/

git clone https://github.com/Freevini/Circleries.git

scp -r /home/vincent/Desktop/Projects/2020_circleries/01_script/random_line.txt vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/

```



## get the SRAs

Here, I first identify 20 SRAs that include long read data don't have a circular genome

```{bash}

##testing loop 
for lineNUM in $(echo "1 7141 2 3 2800 5 6")
do
echo $lineNUM

alreadyyy=$( awk -v linezzz="$lineNUM" '{if($1==linezzz) print $1}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly |wc -l |cut -f 1 )
 #if [ $alreadyyy -gt "0" ]; then exit fi
   if [[ $alreadyyy -gt 0 ]]; then
    continue
  fi
 echo "not yet"

done

##----------------------prep 2
wc -l /data/Project/2021_circlator/NCBI_analysis/random_line.txt
head /data/Project/2021_circlator/NCBI_analysis/random_line.txt

for liness in $(cut -f 1 /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly)
#for liness in $(echo "11881 8098")

do

grep -v "^${liness}$(printf '\n')" /data/Project/2021_circlator/NCBI_analysis/random_line.txt > /data/Project/2021_circlator/NCBI_analysis/random_line_tmp.txt
cat  /data/Project/2021_circlator/NCBI_analysis/random_line_tmp.txt > /data/Project/2021_circlator/NCBI_analysis/random_line.txt
done
head /data/Project/2021_circlator/NCBI_analysis/random_line.txt
wc -l /data/Project/2021_circlator/NCBI_analysis/random_line.txt


##-----------------------main script download and assemble

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/{logs,SRA}


awk '{if($4=="NO")print $0}' /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome > /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular


maxNUM=$(wc -l /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular|cut -d ' ' -f 1)

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names
#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info
#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis


#less /data/Project/2021_circlator/NCBI_analysis/random_line.txt

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/{logs,SRA,flye}
#for count in {1..300}
for lineNUM in $(head -300 /data/Project/2021_circlator/NCBI_analysis/random_line.txt)
do
#echo $lineNUM
#echo $count
#lineNUM=$(echo $((1 + $RANDOM % $maxNUM)))
echo $lineNUM
genomeName=$(sed -n "${lineNUM}p" /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular |cut -f 1)
echo ${genomeName}



#genomeName=GCF_001766355.1_ASM176635v1 # good test case (pacbio, few reads, circular)
#genomeName=GCF_002158885.1_ASM215888v1
grep "${genomeName}$(printf '\t')" /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 3 > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names

SRA=NA

epost -db biosample -input /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info

SRA=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1 |cut -f 1)

/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout ${SRA} > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz
 gzip -c /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz

Pacbio_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|grep "pacbio" -i|wc -l |cut -d ' ' -f 1)

ONT_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|grep "Oxford" -i|wc -l |cut -d ' ' -f 1)

Illumina_yes=$(grep "Illumina" -i  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|wc -l |cut -d ' ' -f 1)

#echo ${Pacbio_yes}
#echo ${ONT_yes}

rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

##-----------------------------------------assemble with flye
mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/

if [ $Pacbio_yes == "1" ]
then
echo "Pacbio data"

python /home/vincent/apps/flye/Flye/bin/flye --threads 37  --iterations 1 --pacbio-raw \
      /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
      --out-dir /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

elif [ $ONT_yes == "1" ]
then
echo "ONT data"

python /home/vincent/apps/flye/Flye/bin/flye --threads 37  --iterations 1 --nano-raw \
      /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
      --out-dir /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

fi

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}/assembly_info.txt  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}/assembly.fasta  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta

rm -r /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}


rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

##final summary

echo -e ${lineNUM}"\t"${genomeName}"\t"${SRA}"\t"${Pacbio_yes}"\t"${ONT_yes}"\t"${Illumina_yes} >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly


done


#cut -f 1,2,3,4,5,6 /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis  > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly



```
# analysis
### check SRAs

```{bash}

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/{logs,SRA,flye}
#for count in {1..300}
for lineNUM in $(tail -12320 /data/Project/2021_circlator/NCBI_analysis/random_line.txt)
do
#echo $lineNUM
#echo $count
#lineNUM=$(echo $((1 + $RANDOM % $maxNUM)))
echo $lineNUM
genomeName=$(sed -n "${lineNUM}p" /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular |cut -f 1)
echo ${genomeName}



#genomeName=GCF_001766355.1_ASM176635v1 # good test case (pacbio, few reads, circular)
#genomeName=GCF_002158885.1_ASM215888v1
genomeName=GCF_016889345.1_ASM1688934v1

grep "${genomeName}$(printf '\t')" /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 3 > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names_tmp


SRA=NA

epost -db biosample -input /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names_tmp -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_02 >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info_02

SRA=$(cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_02 |head -1 |cut -f 1)

#/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout ${SRA} > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz
 #gzip -c /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz



Pacbio_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_02|grep "pacbio" -i|wc -l |cut -d ' ' -f 1)

ONT_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_02 |grep "Oxford" -i|wc -l |cut -d ' ' -f 1)

Illumina_yes=$(grep "Illumina" -i  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_02 |wc -l |cut -d ' ' -f 1)

echo -e ${lineNUM}"\t"${genomeName}"\t"${SRA}"\t"${Pacbio_yes}"\t"${ONT_yes}"\t"${Illumina_yes} >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test

done


cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info_02
cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test

wc -l /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test
awk -F "\t" '{OFS="\t"}{if( $4>0 || $5 > 0) print $0}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test |wc -l

awk -F "\t" '{OFS="\t"}{if( $4 == 1 || $5 == 1) print $0}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test |wc -l


```

### download and assemble

```{bash}


###------------------------------------------------------
#analyse current assemblies
###------------------------------------------------------
rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt
for SRA in $(ls /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/ |grep "_info.txt" |sed 's/_info.txt//g' )
do
#echo $SRA

#cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt

contig_count=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |wc -l)

longest_contig=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 2|head -1)

circ_longest=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|head -1)

num_circ=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|grep "Y"|wc -l |cut -f 1)

num_non_circ=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|grep -v "Y"|wc -l |cut -f 1)

echo -e ${SRA}"\t"${contig_count}"\t"${longest_contig}"\t"${circ_longest}"\t"${num_circ}"\t"${num_non_circ} >>  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt

done

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt

for genomeName in $(awk -F "\t" '{OFS="\t"}{if( $4 == 1 || $5 == 1) print $2}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test |uniq |head -100 )
do

#genomeName=GCF_016889345.1_ASM1688934v1

grep "${genomeName}$(printf '\t')" /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 3 > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names_tmp


#SRA=NA

epost -db biosample -input /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names_tmp -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03 >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info_03

SRA=$(cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp_03|grep -v -i "Illumina" |head -1 |cut -f 1)

echo -e ${SRA}
#SRA=SRR13181905
#SRA=SRR13181905
/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout ${SRA} > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz
 gzip -c /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz &



awk -v srazzz="$SRA" '{if($1==srazzz)print $0}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info_03 > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

Pacbio_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|grep "pacbio" -i|wc -l |cut -d ' ' -f 1)

ONT_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|grep "Oxford" -i|wc -l |cut -d ' ' -f 1)

Illumina_yes=$(grep "Illumina" -i  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|wc -l |cut -d ' ' -f 1)

#echo ${Pacbio_yes}
#echo ${ONT_yes}

rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

##-----------------------------------------assemble with flye
mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/

if [ $Pacbio_yes -gt 0 ]
then
echo "Pacbio data"

python /home/vincent/apps/flye/Flye/bin/flye --threads 37  --iterations 1 --pacbio-raw \
      /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq \
      --out-dir /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

elif [ $ONT_yes -gt 0 ]
then
echo "ONT data"

python /home/vincent/apps/flye/Flye/bin/flye --threads 37  --iterations 1 --nano-raw \
      /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq \
      --out-dir /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

fi

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}/assembly_info.txt  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}/assembly.fasta  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta

rm -r /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

##----------------------final summary

contig_count=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |wc -l)

longest_contig=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 2|head -1)

circ_longest=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|head -1)

num_circ=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|grep "Y"|wc -l |cut -f 1)

num_non_circ=$(grep "^#" -v /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|grep -v "Y"|wc -l |cut -f 1)

echo -e ${SRA}"\t"${contig_count}"\t"${longest_contig}"\t"${circ_longest}"\t"${num_circ}"\t"${num_non_circ} >>  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt


done

SRR10586120

##-------------------------------------delete SRA which are not interesting

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt

#awk -F "\t" '{OFS="\t"}{if($2<20 && $3>800000 && $4=="Y" &&)}'

awk -F "\t" '{OFS="\t"}{if($2>15 && $3<500000) print $0}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt


for SRA in $(awk -F "\t" '{OFS="\t"}{if($2>15 && $3<500000) print $1}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt)
do

echo ${SRA}

rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.*

done 


cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt


awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $0}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt

awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $4}' /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt |sort|uniq -c




```

## circlator vs. circleries

```{bash}
##-----------------------------------------circlator

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/
mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp

rm -r  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/
start=`date +%s`
circlator all --threads 8 /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/

end=`date +%s`
runtime_circlator=$( echo "$end - $start" | bc -l )
echo $runtime_circlator

echo -e ${SRA}"\t"${runtime_circlator}"\tcirclator" >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/runtimes.circlator


cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/06.fixstart.detailed.log /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/${SRA}_fixstart.log

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/04.merge.circularise.log /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/${SRA}_circularise.log

rm -r /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}

##-----------------------------------------circleres

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}
start=`date +%s`

/data/Project/2021_circlator/20220106_2020_circleries_tool/Circleries/Circleries.sh \
    -i /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta \
    -l /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
    -t 8 \
    -o ${SRA} \
    -d /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/ \
    -F 
    
end=`date +%s`
runtime_circleres=$( echo "$end - $start" | bc -l )
echo $runtime_circleres


mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/${SRA}

echo -e ${SRA}"\t"${runtime_circleres}"\tcircleres" >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/runtimes.circleres


cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}_analysis_circularity_extended.log  \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/${SRA}_analysis_circularity_extended.log 
    
cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}.fasta \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/${SRA}.fasta
   
rm -r  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/

rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

##final summary


echo -e ${lineNUM}"\t"${genomeName}"\t"${SRA}"\t"${Pacbio_yes}"\t"${ONT_yes}"\t"${Illumina_yes}"\t"${runtime_circlator}"\t"${runtime_circleres} >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis

done


cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/*_info.txt

cat  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/*_analysis_circularity_extended.log 

cat  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/*_circularise.log


```




# rerun without assembly

```{bash}

for SRA in $(cut -f 3 /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis |grep "SR")
do

echo $SRA
##---------------------------circleries

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}
start=`date +%s`

/data/Project/2021_circlator/20220106_2020_circleries_tool/Circleries/Circleries.sh \
    -i /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta \
    -l /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
    -t 8 \
    -o ${SRA} \
    -d /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/ \
    -F 
    
end=`date +%s`
runtime_circleres=$( echo "$end - $start" | bc -l )
echo $runtime_circleres


mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/

#echo -e ${SRA}"\t"${runtime_circleres}"\tcircleres" >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/runtimes.circleres

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}_analysis_circularity_extended.log  \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/${SRA}_analysis_circularity_extended.log 
    
cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}.fasta \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/${SRA}.fasta
   
rm -r  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/

##final summary

done
```






