---
title: "20211229_circles_analysis"
author: "Vincent Somerville"
date: "30/12/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)
```


## check start alignment

Here, I check how many completely assembled genomes are start aligned on NCBI. 

```{bash}
mkdir -p /data/Project/2021_circlator/NCBI_analysis/log/
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt


cut -f 12 /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt|sort|uniq -c


##------------------------------------------------
#only chromosome
##------------------------------------------------

awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Chromosome" ) print $0}' /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 20 | sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome
    
    
cd 
rm -r /data/Project/2021_circlator/NCBI_analysis/download/gff/
 mkdir -p /data/Project/2021_circlator/NCBI_analysis/download/gff/
 
 rm -r /data/Project/2021_circlator/NCBI_analysis/analysis
 mkdir -p /data/Project/2021_circlator/NCBI_analysis/analysis/
 
 
cd /data/Project/2021_circlator/NCBI_analysis/download/gff/

#wget -i /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome


for downloadsss in $(cat /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome)
do

echo $downloadsss

namess=$(echo $downloadsss |sed 's/.*\///g'|sed 's/_genomic.gff.gz//g')

wget  ${downloadsss}


zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $0}' >> /data/Project/2021_circlator/NCBI_analysis/analysis/genes.identified.chromosome
startss=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $4}')
orientation=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $7}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq" && $4 < "5000" && $7 =="+")print "Yes"; else print "NO"}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq") && ($4 < "5000") && ( $7 =="+")) print "Yes" }')

startAligned=$(echo $startss | awk -v orientations="$orientation" '{if(($1 < 5000) && (orientations == "+")) print "Yes"; else print "NO" }')



#echo -e ${namess}
#echo -e ${namess}"\t"${startss}
#echo -e ${namess}"\t"${startss}"\t"${orientation}
#echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} 

#zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq" && $4 < "5000" && $7 =="+")) print "Yes" }'

#|awk -F "\t" -v contigNames="$contigName" '{OFS="\t"}{if($2=="bacterial_contig"  && ($7=="Y"||$8=="Y"||$9=="Y"))print "Y"; else print "N"}'

echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} >> /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome
#rm ${namess}*
done


cat /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome

cut -f 4 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome |sort|uniq -c
cut -f 3 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome |sort|uniq -c

wc -l /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome
wc -l /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome



##------------------------------------------------
#only complete genome
##------------------------------------------------

awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ) print $0}' /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 20 | sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_Complete_Genome
    
    
cd 
#rm -r /data/Project/2021_circlator/NCBI_analysis/download/gff/
#mkdir -p /data/Project/2021_circlator/NCBI_analysis/download/gff/
 
# rm -r /data/Project/2021_circlator/NCBI_analysis/analysis
# mkdir -p /data/Project/2021_circlator/NCBI_analysis/analysis/
 
 
cd /data/Project/2021_circlator/NCBI_analysis/download/gff/

#wget -i /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_chromosome


for downloadsss in $(cat /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_Complete_Genome)
do

echo $downloadsss

namess=$(echo $downloadsss |sed 's/.*\///g'|sed 's/_genomic.gff.gz//g')

wget  ${downloadsss}

zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $0}' >> /data/Project/2021_circlator/NCBI_analysis/analysis/genes.identified.Complete_Genome
startss=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $4}')
orientation=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $7}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq" && $4 < "5000" && $7 =="+")print "Yes"; else print "NO"}')
#startAligned=$(zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq") && ($4 < "5000") && ( $7 =="+")) print "Yes" }')

startAligned=$(echo $startss | awk -v orientations="$orientation" '{if(($1 < 5000) && (orientations == "+")) print "Yes"; else print "NO" }')



#echo -e ${namess}
#echo -e ${namess}"\t"${startss}
#echo -e ${namess}"\t"${startss}"\t"${orientation}
#echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} 

#zgrep "=dnaA;" -i ${namess}* | awk '{if(($2=="RefSeq" && $4 < "5000" && $7 =="+")) print "Yes" }'

#|awk -F "\t" -v contigNames="$contigName" '{OFS="\t"}{if($2=="bacterial_contig"  && ($7=="Y"||$8=="Y"||$9=="Y"))print "Y"; else print "N"}'

echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} >> /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome
#rm ${namess}*
done


cat /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome

cut -f 4 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome |sort|uniq -c
cut -f 3 /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome |sort|uniq -c

wc -l /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome
wc -l /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.download_Complete_Genome


##----------------------------------------------------
#move local

mkdir -p  /home/vincent/Desktop/Projects/2020_circleries/07_figures
mkdir -p  /home/vincent/Desktop/Projects/2020_circleries/06_analysis/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.chromosome /home/vincent/Desktop/Projects/2020_circleries/06_analysis/

mkdir -p  /home/vincent/Desktop/Projects/2020_circleries/06_analysis/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome /home/vincent/Desktop/Projects/2020_circleries/06_analysis/


```


```{r}


library(tidyverse)

#-----------------------------------------------------------------------
#chromosome
#-----------------------------------------------------------------------

dnaA_genes_stats <- read_delim("~/Desktop/Projects/2020_circleries/06_analysis/dnaA_genes.stats.chromosome", delim = "\t", escape_double = FALSE, col_names = c("genome","startpostion","orientation","startAligned"), trim_ws = TRUE)

plot_chromsome <- ggplot(dnaA_genes_stats,aes(x=startAligned,fill=startAligned,color=startAligned))+geom_bar()+theme_classic()+labs(x="Genome start aligned\n at dnaA",y="Genomes",title = "Chromosome-level assemblies\non NCBI")+theme(legend.position = "none")

png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_chromosome_startAligned_NCBI.png", width = 2200, height = 3000,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plot_chromsome

dev.off()


#-----------------------------------------------------------------------
#complete genomes
#-----------------------------------------------------------------------

dnaA_genes_stats_complete <- read_delim("~/Desktop/Projects/2020_circleries/06_analysis/dnaA_genes.stats.Complete_Genome", delim = "\t", escape_double = FALSE, col_names = c("genome","startpostion","orientation","startAligned"), trim_ws = TRUE)

plot_complete_genome <- ggplot(dnaA_genes_stats_complete,aes(x=startAligned,fill=startAligned,color=startAligned))+geom_bar()+theme_classic()+labs(x="Genome start aligned\n at dnaA",y="Genomes",title = "Complete genome assemblies\non NCBI")+theme(legend.position = "none")
plot_complete_genome
png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_completeGenomes_startAligned_NCBI.png", width = 2200, height = 3000,res=400)
# pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)

plot_complete_genome

dev.off()


#-----------------------------------------------------------------------
#together
#-----------------------------------------------------------------------

library(patchwork)
plot_complete_genome+plot_chromsome

png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_both_startAligned_NCBI.png", width = 4400, height = 3000,res=400)

plot_complete_genome+plot_chromsome

dev.off()

```


# biosample2SRA

download SRA files to test circlator and circleries on 20 long read data. 

from https://www.biostars.org/p/448799/ 

```{bash}
##--------------------------------------------------------
#test
##--------------------------------------------------------
cd /data/Project/2021_circlator/NCBI_analysis/log/
cat samples.txt 
SAMN14390563
SAMN14390566
SAMN14390576
SAMN14390578
SAMN14453547
SAMN14453553


epost -db biosample -input samples.test -format acc | \
elink -target sra | \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > samples.test.sra


/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout SRR390728 

/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout SRR390728 

SAMN14390563

```



# move circleries to bigmachine

```{bash}




scp -r /home/vincent/Desktop/Projects/2020_circleries vincent@130.223.51.66:/data/Project/2021_circlator/20220106_2020_circleries_tool/

cd /data/Project/2021_circlator/20220106_2020_circleries_tool/

git clone https://github.com/Freevini/Circleries.git

```



## get the SRAs

Here, I first identify 20 SRAs that include long read data don't have a circular genome

```{bash}

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/{logs,SRA}


awk '{if($4=="NO")print $0}' /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome > /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular


maxNUM=$(wc -l /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular|cut -d ' ' -f 1)

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names
#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info
#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis


mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/{logs,SRA,flye}
for count in {1..200}
do

echo $count
lineNUM=$(echo $((1 + $RANDOM % $maxNUM)))
echo $lineNUM
genomeName=$(sed -n "${lineNUM}p" /data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular |cut -f 1)
echo ${genomeName}

#genomeName=GCF_001766355.1_ASM176635v1 # good test case (pacbio, few reads, circular)
#genomeName=GCF_002158885.1_ASM215888v1
grep "${genomeName}$(printf '\t')" /data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt |cut -f 3 > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names

SRA=NA

epost -db biosample -input /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/SRA_biosample.names.sra.info

SRA=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1 |cut -f 1)

/home/vincent/apps/sratoolkit.2.11.2-ubuntu64/bin/fastq-dump --stdout ${SRA} > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

#rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz
 gzip -c /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq > /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz

Pacbio_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|grep "pacbio" -i|wc -l |cut -d ' ' -f 1)

ONT_yes=$(grep "Illumina" -i -v  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|grep "Oxford" -i|wc -l |cut -d ' ' -f 1)

Illumina_yes=$(grep "Illumina" -i  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp |head -1|wc -l |cut -d ' ' -f 1)




#echo ${Pacbio_yes}
#echo ${ONT_yes}

rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/logs/tmp

##-----------------------------------------assemble with flye
mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/

if [ $Pacbio_yes == "1" ]
then
echo "Pacbio data"

python /home/vincent/apps/flye/Flye/bin/flye --threads 37  --iterations 1 --pacbio-raw \
      /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
      --out-dir /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

elif [ $ONT_yes == "1" ]
then
echo "ONT data"

python /home/vincent/apps/flye/Flye/bin/flye --threads 37  --iterations 1 --nano-raw \
      /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
      --out-dir /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

fi

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}/assembly_info.txt  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}_info.txt

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}/assembly.fasta  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta

rm -r /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/tmp/${SRA}

##-----------------------------------------circlator

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/
mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp

rm -r  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/
start=`date +%s`
circlator all --threads 8 /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/

end=`date +%s`
runtime_circlator=$( echo "$end - $start" | bc -l )
echo $runtime_circlator

echo -e ${SRA}"\t"${runtime_circlator}"\tcirclator" >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/runtimes.circlator


cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/06.fixstart.detailed.log /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/${SRA}_fixstart.log

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}/04.merge.circularise.log /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/${SRA}_circularise.log

rm -r /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/tmp/${SRA}

##-----------------------------------------circleres

mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}
start=`date +%s`

/data/Project/2021_circlator/20220106_2020_circleries_tool/Circleries/Circleries.sh \
    -i /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta \
    -l /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
    -t 8 \
    -o ${SRA} \
    -d /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/ \
    -F 
    
end=`date +%s`
runtime_circleres=$( echo "$end - $start" | bc -l )
echo $runtime_circleres


mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/${SRA}

echo -e ${SRA}"\t"${runtime_circleres}"\tcircleres" >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/runtimes.circleres


cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}_analysis_circularity_extended.log  \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/${SRA}_analysis_circularity_extended.log 
    
cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}.fasta \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/${SRA}.fasta
   
rm -r  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/

rm /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq

##final summary


echo -e ${lineNUM}"\t"${genomeName}"\t"${SRA}"\t"${Pacbio_yes}"\t"${ONT_yes}"\t"${Illumina_yes}"\t"${runtime_circlator}"\t"${runtime_circleres} >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis

done


cat /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/*_info.txt

cat  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered/*_analysis_circularity_extended.log 

cat  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circlator/all_gathered/*_circularise.log


```

# rerun without assembly


```{bash}

for SRA in $(cut -f 3 /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis)
do


##---------------------------circleries


mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}
start=`date +%s`

/data/Project/2021_circlator/20220106_2020_circleries_tool/Circleries/Circleries.sh \
    -i /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/${SRA}.fasta \
    -l /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/SRA/${SRA}.fastq.gz \
    -t 8 \
    -o ${SRA} \
    -d /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/ \
    -F 
    
end=`date +%s`
runtime_circleres=$( echo "$end - $start" | bc -l )
echo $runtime_circleres


mkdir -p /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/

#echo -e ${SRA}"\t"${runtime_circleres}"\tcircleres" >> /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/runtimes.circleres

cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}_analysis_circularity_extended.log  \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/${SRA}_analysis_circularity_extended.log 
    
cp /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/${SRA}.fasta \
    /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/all_gathered_again/${SRA}.fasta
   
rm -r  /data/Project/2021_circlator/NCBI_analysis/SRA_analysis/circleres/tmp/${SRA}/

##final summary

done
```






