---
title: "Supplement: ReCycled: A tool to restart circular bacterial chromosomes"
author: "Vincent Somerville"
date: "23/1/2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=TRUE, eval=FALSE)
```

This is the supplemental analysis of the application note:

*ReCycled: A tool to restart circular bacterial chromosomes*

# Data deposit

All data necessary to reproudce the analysis  is deposited on (Zonedo)[]. 
In order to rerun the analysis the files need to stored locally and the directories should be adjusted here. 
the analysis_location should be the directory which contains the downloaded files. Further the analysis will produce new files and directories therein. 

```{bash}
analysis_location="/change/the/location/here"
```

also change the R working directory similiarly.
```{r eval=TRUE}
#knitr::opts_knit$set(root.dir = '/change/the/location/here')
#setwd("/change/the/location/here")

# e.g. for my computer
knitr::opts_knit$set(root.dir = '/home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/')
setwd("/home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/")
```

# check start alignment

Here, we checked how many completely assembled genomes are start aligned on NCBI. 
We therefore scanned all NCBI RegSeq genomes that are flagged as *complete assembly* if dnaA is located within the first 5000bp and if it is on the positive strand. 

```{bash echo=FALSE}
mkdir -p /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/log/20211229_ncbi.txt /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/analysis/dnaA_genes.stats.Complete_Genome_non_circular /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_assembly_final_test /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/NCBI_RefSeq_sequencingData_availableOnSRA
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/SRA_analysis/flye/all_gathered/genome_analysis.txt /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/genome_analysis.txt
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/SRA_analysis/startaligned_analysis.final /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/restart_analysis.final
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_02 /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/RunTime_Circlator.txt
scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/SRA_analysis/overall_analysis_03_onlycirclers /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/RunTime_ReCycled.txt

scp -r vincent@130.223.51.66:/data/Project/2021_circlator/NCBI_analysis/SRA_analysis/Circ_contig_analysis.final /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/Circularity_of_bacterial_contig.analysis

cp /home/vincent/Desktop/Projects/2020_circleries/06_analysis/scripts/random_line.txt /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/
cp /home/vincent/Desktop/Projects/2020_circleries/06_analysis/dnaA_genes.stats.Complete_Genome  /home/vincent/Desktop/Projects/2020_circleries/08_Zenodo_files/
```


```{bash}
##------------------------------------------------
#downloaded the NCBI refseq assembly_summary 
#---------------------THIS FILE IS STORED ON ZONEDO------------------------------------
##------------------------------------------------

mkdir -p ${analysis_location}
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > ${analysis_location}/20211229_ncbi.txt
cut -f 12 ${analysis_location}/20211229_ncbi.txt|sort|uniq -c

##------------------------------------------------
#only complete genome
##------------------------------------------------

awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ) print $0}' ${analysis_location}/20211229_ncbi.txt |cut -f 20 | sed -r 's|(https://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    ${analysis_location}//20211229_ncbi.download_Complete_Genome

mkdir -p ${analysis_location}/download/gff/
cd ${analysis_location}/download/gff/

for downloadsss in $(cat ${analysis_location}//20211229_ncbi.download_Complete_Genome)
do

echo $downloadsss
namess=$(echo $downloadsss |sed 's/.*\///g'|sed 's/_genomic.gff.gz//g')

wget  ${downloadsss}

zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $0}' >> ${analysis_location}/genes.identified.Complete_Genome
startss=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $4}')
orientation=$(zgrep "=dnaA;" -i ${namess}* | awk '{if($2=="RefSeq")print $7}')

startAligned=$(echo $startss | awk -v orientations="$orientation" '{if(($1 < 5000) && (orientations == "+")) print "Yes"; else print "NO" }')

echo -e ${namess}"\t"${startss}"\t"${orientation}"\t"${startAligned} >> ${analysis_location}/dnaA_genes.stats.Complete_Genome
done

cat  ${analysis_location}/dnaA_genes.stats.Complete_Genome

cut -f 4  ${analysis_location}/dnaA_genes.stats.Complete_Genome |sort|uniq -c
cut -f 3  ${analysis_location}/dnaA_genes.stats.Complete_Genome |sort|uniq -c

wc -l  ${analysis_location}/dnaA_genes.stats.Complete_Genome
wc -l  ${analysis_location}/20211229_ncbi.download_Complete_Genome

#---------------------THIS FILE IS STORED ON ZONEDO------------------------------------

awk '{if($4=="NO")print $0}' ${analysis_location}/dnaA_genes.stats.Complete_Genome >  ${analysis_location}/dnaA_genes.stats.Complete_Genome_non_circular
```

Here we analyse the data in R. 

```{r eval=TRUE}
library(tidyverse)
#-----------------------------------------------------------------------
#complete genomes
#-----------------------------------------------------------------------
dnaA_genes_stats_complete <- read_delim("dnaA_genes.stats.Complete_Genome", delim = "\t", escape_double = FALSE, col_names = c("genome","startpostion","orientation","startAligned"), trim_ws = TRUE)

table(dnaA_genes_stats_complete$startAligned)

plot_complete_genome <- ggplot(dnaA_genes_stats_complete,aes(x=startAligned,fill=startAligned,color=startAligned))+geom_bar()+theme_classic()+labs(x="Bacterial chromosome start aligned",y="Genomes",title = "Complete genome assemblies\non NCBI efSeq")+theme(legend.position = "none")+scale_y_continuous(breaks=c(0,5000,11496,13015)) 
plot_complete_genome

# png("/home/vincent/Desktop/Projects/2020_circleries/07_figures/Plot_completeGenomes_startAligned_NCBI.png", width = 2200, height = 3000,res=400)
# # pdf("~/Desktop/Projects/2020_StarterCultureDiversity/97_Images/Sterm_phylogney_total.pdf",width=10,height=10)
# 
# plot_complete_genome
# 
# dev.off()
```

# Download ReCycled

This is the download of (Recycled)[https://github.com/Freevini/ReCycled].

```{bash}
git clone https://github.com/Freevini/ReCycled.git
```

#  Download Circlator

This is the download of (Circlator)[https://github.com/sanger-pathogens/circlator] (Hunt et al. 2015). 

```{bash}
pip3 install circlator
```

## Randomize SRA for download

Here, I first randomize the order of the identified SRAs that include long read data but have do not have restarted genomes.
I do this in order to then download SRA and restart them with the different tools. 

I first sample random numbers in R.  

```{r}
library(tidyverse)
random_line <- sample(1:12321, 12321, replace=FALSE)

unique(random_line) %>% length

write_csv(random_line,"random_line.txt")
```

Afterwards I assign the random numbers to the SRAs. 

```{bash}
wc -l ${analysis_location}/random_line.txt
head ${analysis_location}/random_line.txt

for liness in $(cut -f 1  ${analysis_location}/overall_analysis_assembly)
do

grep -v "^${liness}$(printf '\n')" ${analysis_location}/random_line.txt > ${analysis_location}/random_line_tmp.txt
cat ${analysis_location}/random_line_tmp.txt > ${analysis_location}/random_line.txt
done
head ${analysis_location}/random_line.txt
wc -l ${analysis_location}/random_line.txt

#---------------------random_line.txt IS STORED ON ZONEDO------------------------------------
```

# check SRAs

Not all NCBI RefSeq entries have a linked SRA and if they do not all SRA contain long read data. 
Therefore I first checked all non-restarted RefSeq genomes if they have SRAs containing long read data (Oxford Nanopore or Pacbio).
If they fulfill these requirements, the genomes and the corresponds SRAs can be used for the benchmarking. 

```{bash}
mkdir -p ${analysis_location}//SRA_analysis/{logs,SRA,flye}
for lineNUM in $(cat ${analysis_location}/random_line.txt)
do

echo $lineNUM
genomeName=$(sed -n "${lineNUM}p" ${analysis_location}/dnaA_genes.stats.Complete_Genome_non_circular |cut -f 1)
echo ${genomeName}

grep "${genomeName}$(printf '\t')" ${analysis_location}/20211229_ncbi.txt |cut -f 3 > ${analysis_location}/SRA_biosample.names_tmp

SRA=NA

epost -db biosample -input ${analysis_location}/SRA_biosample.names_tmp -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent >  ${analysis_location}/SRA_analysis/logs/tmp_03

cat ${analysis_location}/SRA_analysis/logs/tmp_03
cat ${analysis_location}/SRA_analysis/logs/tmp_03 >> ${analysis_location}/SRA_analysis/logs/SRA_biosample.names.sra.info_02

SRA=$(cat ${analysis_location}/SRA_analysis/logs/tmp_032 |head -1 |cut -f 1)
Pacbio_yes=$(grep "Illumina" -i -v ${analysis_location}/SRA_analysis/logs/tmp_03 |grep "pacbio" -i|wc -l |cut -d ' ' -f 1)
ONT_yes=$(grep "Illumina" -i -v  ${analysis_location}/SRA_analysis/logs/tmp_03 |grep "Oxford" -i|wc -l |cut -d ' ' -f 1)

Illumina_yes=$(grep "Illumina" -i ${analysis_location}/SRA_analysis/logs/tmp_03 |wc -l |cut -d ' ' -f 1)
echo -e ${lineNUM}"\t"${genomeName}"\t"${SRA}"\t"${Pacbio_yes}"\t"${ONT_yes}"\t"${Illumina_yes} >> ${analysis_location}/NCBI_RefSeq_sequencingData_availableOnSRA

done

cat ${analysis_location}/NCBI_RefSeq_sequencingData_availableOnSRA

wc -l ${analysis_location}/NCBI_RefSeq_sequencingData_availableOnSRA
awk -F "\t" '{OFS="\t"}{if( $4>0 || $5 > 0) print $0}' ${analysis_location}/NCBI_RefSeq_sequencingData_availableOnSRA |wc -l
awk -F "\t" '{OFS="\t"}{if( $4 == 1 || $5 == 1) print $0}' ${analysis_location}/NCBI_RefSeq_sequencingData_availableOnSRA |wc -l

#---------------------NCBI_RefSeq_sequencingData_availableOnSRA IS STORED ON ZONEDO------------------------------------
```

From all 12321 non-restarted NCBI RefSeq genomes only 2813 fulfill the requirement of containing long read SRAs. 
I will now proceed with a random subsample of these. 

# Download and assemble

Here I download the SRA by using the NCBI toolbox (E-utils)[https://www.ncbi.nlm.nih.gov/books/NBK25499/]. 
Aftwards I assemble the raw read data with (flye)[https://github.com/fenderglass/Flye] (Kolmogorov et al. 2019)


```{bash}
###------------------------------------------------------
#subsample certain genomes and download/assemble accordingly
###------------------------------------------------------

mkdir -p ${analysis_location}/SRA_analysis/logs/

for genomeName in $(awk -F "\t" '{OFS="\t"}{if( $4 == 1 || $5 == 1) print $2}' ${analysis_location}/NCBI_RefSeq_sequencingData_availableOnSRA |uniq |head -600 |tail -299 )
do

grep "${genomeName}$(printf '\t')" ${analysis_location}/20211229_ncbi.txt |cut -f 3 > ${analysis_location}/SRA_analysis/logs/SRA_biosample.names_tmp

epost -db biosample -input ${analysis_location}/SRA_analysis/logs/SRA_biosample.names_tmp -format acc | \
elink -target sra| \
efetch -db sra -format runinfo -mode xml | \
xtract -pattern Row -def "NA" -element Run spots bases spots_with_mates avgLength \
size_MB download_path Experiment LibraryStrategy LibrarySelection LibrarySource \
LibraryLayout InsertSize InsertDev Platform Model SRAStudy BioProject ProjectID \
Sample BioSample SampleType TaxID ScientificName SampleName CenterName \
Submission Consent > ${analysis_location}/SRA_analysis/logs/tmp_03

cat ${analysis_location}/SRA_analysis/logs/tmp_03 >>  ${analysis_location}/SRA_analysis/logs/SRA_biosample.names.sra.info_03
SRA=$(cat ${analysis_location}/SRA_analysis/logs/tmp_03|grep -v -i "Illumina" |head -1 |cut -f 1)
mkdir -p ${analysis_location}/SRA_analysis/SRA

echo -e ${SRA}

##-----------------------------------------downlad fastq

fastq-dump --stdout ${SRA} > ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq
gzip -c ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq > ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq.gz &

awk -v srazzz="$SRA" '{if($1==srazzz)print $0}' ${analysis_location}/SRA_analysis/logs/SRA_biosample.names.sra.info_03 > ${analysis_location}/SRA_analysis/logs/tmp
Pacbio_yes=$(grep "Illumina" -i -v  ${analysis_location}/SRA_analysis/logs/tmp |head -1|grep "pacbio" -i|wc -l |cut -d ' ' -f 1)
ONT_yes=$(grep "Illumina" -i -v  ${analysis_location}/SRA_analysis/logs/tmp|head -1|grep "Oxford" -i|wc -l |cut -d ' ' -f 1)
Illumina_yes=$(grep "Illumina" -i ${analysis_location}/SRA_analysis/logs/tmp |head -1|wc -l |cut -d ' ' -f 1)

##-----------------------------------------assemble with flye
mkdir -p ${analysis_location}/SRA_analysis/flye/tmp/

if [ $Pacbio_yes -gt 0 ]
then
echo "Pacbio data"

flye --threads 37  --iterations 1 --pacbio-raw \
      ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq \
      --out-dir ${analysis_location}/SRA_analysis/flye/tmp/${SRA}

elif [ $ONT_yes -gt 0 ]
then
echo "ONT data"

flye --threads 37  --iterations 1 --nano-raw \
      ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq \
      --out-dir ${analysis_location}/SRA_analysis/flye/tmp/${SRA}

fi

mkdir -p ${analysis_location}/SRA_analysis/flye/all_gathered

cp ${analysis_location}/SRA_analysis/flye/tmp/${SRA}/assembly_info.txt  ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}_info.txt
cp ${analysis_location}/SRA_analysis/flye/tmp/${SRA}/assembly.fasta  ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}.fasta

rm -r  ${analysis_location}/SRA_analysis/flye/tmp/${SRA}

##----------------------final summary

contig_count=$(grep "^#" -v ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}_info.txt |wc -l)
longest_contig=$(grep "^#" -v ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 2|head -1)
circ_longest=$(grep "^#" -v ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|head -1)
num_circ=$(grep "^#" -v ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|grep "Y"|wc -l |cut -f 1)
num_non_circ=$(grep "^#" -v ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}_info.txt |sort -k2 -n -r | cut -f 4|grep -v "Y"|wc -l |cut -f 1)
echo -e ${SRA}"\t"${contig_count}"\t"${longest_contig}"\t"${circ_longest}"\t"${num_circ}"\t"${num_non_circ} >>   ${analysis_location}/genome_analysis.txt

done

#---------------------genome_analysis.txt IS STORED ON ZONEDO------------------------------------
```

# Qualtiy control of downlaod and assembly

After the download and assembly, I checked the quality of the assembly. 
Some assemblies are very fragmented. I therefore filter out all assemblies that have either more then 15 contigs or the largest contig is smaller then 500 kb. 
The reason why some assemblies are very fragmented is that not all data has been deposited or downloaded. 


```{bash}
##-------------------------------------------------------------------------
##-------------------------------------delete SRA which are not interesting
#to save space
##-------------------------------------------------------------------------
cat ${analysis_location}/genome_analysis.txt

awk -F "\t" '{OFS="\t"}{if($2>15 && $3<500000) print $0}' ${analysis_location}/genome_analysis.txt

for SRA in $(awk -F "\t" '{OFS="\t"}{if($2>15 && $3<500000) print $1}' ${analysis_location}/genome_analysis.txt)
do

echo ${SRA}

rm ${analysis_location}/SRA_analysis/SRA/${SRA}*

done 

##-------------------------------------------------------------------------
##-------------------------------------make a list of SRA to analyse with circlator and Recycled
##-------------------------------------------------------------------------

awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $0}' ${analysis_location}/genome_analysis.txt
awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $4}' ${analysis_location}/genome_analysis.txt |sort|uniq -c

##----------------------SRA to analyse----------------------

awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $1,$4}' ${analysis_location}/genome_analysis.txt |sort|uniq |cut -f 2|sort|uniq -c
awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $1}' ${analysis_location}/genome_analysis.txt |sort|uniq  > ${analysis_location}/genome_analysis_toDo.txt 
wc -l ${analysis_location}/genome_analysis_toDo.txt 

```

## Circularisation analysis

Now I use the two circlatriation and restart tools to compare on the selected data. 

### Circlator 

I use Circlator with default parameters and 8 threads. 

```{bash}
##==============================================================
# ONLY CIRCLATOR
##=============================================================
tmpdir=tmp_08

for SRA in $(cat ${analysis_location}/genome_analysis_toDo.txt |sort|uniq |sed -n 33,42p )
do
echo ${SRA}
##-----------------------------------------circlator

mkdir -p ${analysis_location}/SRA_analysis/circlator/all_gathered_02/
mkdir -p ${analysis_location}/SRA_analysis/circlator/${tmpdir}

rm -r  /${tmpdir}/${tmpdir}/${SRA}/
start=`date +%s`
circlator all --threads 8 / ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}.fasta  ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq  ${analysis_location}/SRA_analysis/circlator/${tmpdir}/${SRA}/

end=`date +%s`
runtime_circlator=$( echo "$end - $start" | bc -l )
echo $runtime_circlator

cp  ${analysis_location}/SRA_analysis/circlator/${tmpdir}/${SRA}/06.fixstart.detailed.log  ${analysis_location}/SRA_analysis/circlator/all_gathered_02/${SRA}_fixstart.log
cp  ${analysis_location}/SRA_analysis/circlator/${tmpdir}/${SRA}/04.merge.circularise.log  ${analysis_location}/SRA_analysis/circlator/all_gathered_02/${SRA}_circularise.log

##-------------------------------final summary

echo -e ${SRA}"\t"${runtime_circlator}"\tNA" >>  ${analysis_location}/RunTime_Circlator.txt

done

#---------------------RunTime_Circlator.txt IS STORED ON ZONEDO------------------------------------
```

### ReCycled

Second I use ReCyled with default paramters and 8 threads. 

```{bash}
##==============================================================
#ONLY ReCycled
##==============================================================
conda activate ReCycled

tmpdir=tmp_08
rm ${analysis_location}/RunTime_ReCycled.txt
for SRA in $(cat ${analysis_location}/SRA_analysis/flye/all_gathered/genome_analysis_toDo.txt|sort|uniq |sed -n 33,42p )
do

##-------------------------------ReCycled

mkdir -p ${analysis_location}/SRA_analysis/ReCycled/${tmpdir}/${SRA}
start=`date +%s`

ReCycled.sh \
    -i ${analysis_location}/SRA_analysis/flye/all_gathered/${SRA}.fasta \
    -l ${analysis_location}/SRA_analysis/SRA/${SRA}.fastq \
    -t 8 \
    -o ${SRA} \
    -d ${analysis_location}/SRA_analysis/ReCycled/${tmpdir}/${SRA}/ \
    -F -x

end=`date +%s`
runtime_circleres=$( echo "$end - $start" | bc -l )
echo $runtime_circleres

mkdir -p ${analysis_location}/SRA_analysis/ReCycled/all_gathered_02/${SRA}


cp ${analysis_location}/SRA_analysis/ReCycled/${tmpdir}/${SRA}/${SRA}_analysis_circularity_extended.log  \
    ${analysis_location}/SRA_analysis/ReCycled/all_gathered_02/${SRA}_analysis_circularity_extended.log 
    
cp ${analysis_location}/SRA_analysis/ReCycled/${tmpdir}/${SRA}/${SRA}.fasta \
    ${analysis_location}/SRA_analysis/ReCycled/all_gathered_02/${SRA}.fasta
 
##-------------------------------final summary
  
echo -e ${SRA}"\tNA\t"${runtime_circleres} >> ${analysis_location}/overall_analysis_03_onlyReCycled

done

wc -l ${analysis_location}/RunTime_ReCycled.txt
cut -f 1 ${analysis_location}/RunTime_ReCycled.txt |sort|uniq |wc -l
awk '{if($2>10)print $1}' ${analysis_location}/overall_analysis_03_onlyReCycled|sort|uniq|wc -l

#---------------------RunTime_ReCycled.txt IS STORED ON ZONEDO------------------------------------
```

# Analysis of outcome

Finally, after I have analysed all genomes with the two methods, I want to answer the following three questions:

1. Are circular contigs recognized?
2. Are the bacterial contigs correctly restarted?
3. How fast are the two tools?

## 1. Are circular contigs recognized?

Here, I check three things. 

1. What is the contig ground truth. Has flye assembled a circular contig?
2. What has Circlator found? Is the contig circular? Is the contig restarted?
3. What has ReCycled found? Is the contig circular? Is the contig restarted?

```{bash}
##----------------------------------------------
#run analysis
##----------------------------------------------

rm ${analysis_location}/restart_analysis.final

for SRA in $(awk '{if($2>10)print $1}' ${analysis_location}/RunTime_Circlator.txt|sort|uniq)
do
echo "==========================="
echo ${SRA}

##------check flye assembly graph (ground truth)

contigzz=$(grep "^#" -v ${analysis_location}//SRA_analysis/flye/all_gathered/${SRA}_info.txt | sort -k2 -n -r |head -1 |cut -f 1)
contigzz_verdict=$(grep "^#" -v ${analysis_location}//SRA_analysis/flye/all_gathered/${SRA}_info.txt | sort -k2 -n -r |head -1 |cut -f 4)

echo -e "--------circlator outcome"

circlator_circ=$(grep "${contigzz}$(printf '\t')" ${analysis_location}//SRA_analysis/circlator/all_gathered_02/${SRA}_circularise.log |cut -f 6)
circlator_start=$(grep "${contigzz}$(printf '\t')" ${analysis_location}//SRA_analysis/circlator/all_gathered_02/${SRA}_fixstart.log |cut -f 7)

if [ $circlator_circ  ==  1 ] &&  [ $circlator_start  =  "-" ]; then
circlator_final=$(echo "Y")
echo "startaligned with Circlator"
elif  [ $circlator_circ  ==  0 ] ||  [ $circlator_start  =  "skipped" ]
then
circlator_final=$(echo "N")
else 
circlator_final=$(echo "NA")
echo "non-startaligned with Circlator"
fi

echo -e "--------circles outcome"

circleres_circ=$(grep "^${contigzz}$(printf '\t')" ${analysis_location}//SRA_analysis/circleres/all_gathered_02/${SRA}_analysis_circularity_extended.log |cut -f 12)

echo -e "--------analysis total"

echo -e ${SRA}"\t"${contigzz_verdict}"\t"${circlator_final}"\t"${circleres_circ} >> ${analysis_location}//restart_analysis.final

done

cat ${analysis_location}//restart_analysis.final
sort ${analysis_location}//restart_analysis.final |uniq | cut -f 2,3,4 |sort |uniq -c
grep -v "NA" ${analysis_location}/restart_analysis.final |sort|uniq

#---------------------restart_analysis.final IS STORED ON ZONEDO------------------------------------
```

## 2. Are the bacterial contigs correctly restarted?

Here, I check the circularity bacterial chromosomes and if they have been restarted. 

1. Which contig is the bacterial chromosome identified with Flye?
2. What has Circlator found? Is the bacterial chromosome circular? Is the bacterial chromosome restarted?
3. What has ReCycled found? Is the bacterial chromosome circular? Is the bacterial chromosome restarted?

```{bash}
echo -e "SRA\tcontig\tflye\tcirclator\tReCycled" > ${analysis_location}/Circularity_of_bacterial_contig.analysis
#for SRA in $(awk -F "\t" '{OFS="\t"}{if($2!=0 && ($2<15 || $3>500000)) print $1}' ${analysis_location}/SRA_analysis/flye/all_gathered/genome_analysis.txt |head -68)

for SRA in $(awk '{if($2>10)print $1}' ${analysis_location}/RunTime_Circlator.txt|sort|uniq)
do
echo "==========================="
echo ${SRA}
##------flye

grep "^#" -v ${analysis_location}//SRA_analysis/flye/all_gathered/${SRA}_info.txt | sort -k1 -n   |cut -f 1,4 > ${analysis_location}//SRA_analysis/tmp.flye

echo -e "--------circlator"

grep -v "#Contig" ${analysis_location}//SRA_analysis/circlator/all_gathered_02/${SRA}_circularise.log | sort -k2 -n |cut -f 2,6 > ${analysis_location}//SRA_analysis/tmp.circlator

echo -e "--------ReCylced"

grep -v "^#contigName" ${analysis_location}//SRA_analysis/circleres/all_gathered_02/${SRA}_analysis_circularity_extended.log | sort -k1 -n   |cut -f 1,8 > ${analysis_location}//SRA_analysis/tmp.circleries

echo -e "--------analysis total"

  for contigss in $(cut -f 1 ${analysis_location}//SRA_analysis/tmp.flye)
  do

flye_circ=$(grep "^${contigss}$(printf '\t')" ${analysis_location}//SRA_analysis/tmp.flye |cut -f 2)
circlator_circ=$(grep "^${contigss}$(printf '\t')" ${analysis_location}/SRA_analysis/tmp.circlator |cut -f 2 |sed 's/0/N/g'|sed 's/1/Y/g')
circleres_circ=$(grep "^${contigss}$(printf '\t')" ${analysis_location}/SRA_analysis/tmp.circleries |cut -f 2)

echo -e ${SRA}"\t"${contigss}"\t"${flye_circ}"\t"${circlator_circ}"\t"${circleres_circ} >> ${analysis_location}/Circularity_of_bacterial_contig.analysis

  done
done

cat ${analysis_location}/Circularity_of_bacterial_contig.analysis
sort ${analysis_location}/Circularity_of_bacterial_contig.analysis |uniq |cut -f 3,5 |sort|uniq -c
sort ${analysis_location}/Circularity_of_bacterial_contig.analysis |uniq |cut -f 3,4,5 |sort|uniq -c

#---------------------Circularity_of_bacterial_contig.analysis IS STORED ON ZONEDO------------------------------------
```

## final analysis

Finally, I bring all the outputs together and look at the data in R. 

```{r eval=TRUE}
library(tidyverse)

##---------------------
#0. genome assemblies on NCBI
##---------------------

dnaA_genes_stats <- read_delim("dnaA_genes.stats.Complete_Genome", delim = "\t", escape_double = FALSE, col_names = c("genome","startpostion","orientation","startAligned"), trim_ws = TRUE)

dnaA_genes_stats$startAligned = factor(dnaA_genes_stats$startAligned, levels=c("Yes","NO"))

colorss <- (c("#BBBE64","#8E5572"))

plot_complete_genome <- ggplot(dnaA_genes_stats,aes(x=startAligned,fill=startAligned,color=startAligned))+geom_bar()+theme_classic()+labs(x="Genome restarted\n at dnaA",y="Genomes",title = "Complete assemblies\non NCBI RefSeq")+theme(legend.position = "none")+scale_fill_manual(values=colorss)+scale_color_manual(values=colorss)+scale_y_continuous(breaks=c(0,5000,11496,13015)) 
# plot_complete_genome

##---------------------
#1. circular bacterial cells aligned
##---------------------

startaligned_analysis <- read_delim("restart_analysis.final", delim = "\t", escape_double = FALSE,  trim_ws = TRUE,col_names = c("SRA","flye","Circlator","ReCycled")) %>% filter(flye=="Y") %>% replace(is.na(.), "N") %>% filter(!is.na(Circlator))
startaligned_analysis <- startaligned_analysis[!duplicated(startaligned_analysis),]

startaligned_analysis_long <- gather(startaligned_analysis, circulariation, outcome, c("Circlator","ReCycled"), factor_key=TRUE,na.rm = TRUE) %>% filter(!is.na(outcome))
startaligned_analysis_long$verdict <- ifelse(startaligned_analysis_long$flye==startaligned_analysis_long$outcome,"restarted","not restarted")
totl_counts <- startaligned_analysis_long %>% group_by(circulariation) %>% summarise(total=n())
startaligned_analysis_sum <- startaligned_analysis_long %>% group_by(circulariation,verdict) %>% summarise(grouping=n()) %>% left_join(.,totl_counts,by="circulariation") %>% mutate(percent=100*(grouping/total))

colorss <- rev(c("#BBBE64","#8E5572"))

countsGenomes <- startaligned_analysis_long$SRA %>% unique() %>% length()
plot_start <- ggplot(startaligned_analysis_sum,aes(x=circulariation,y=percent,fill=verdict,color=verdict))+geom_bar(stat = "identity")+theme_classic()+scale_fill_manual(values=colorss)+scale_color_manual(values=colorss)+
  labs(y="Genomes [%]",title = paste0("Restart of bacterial\nchromosomes (n=",countsGenomes,")"))+
  theme(legend.title = element_blank(),axis.title.x = element_blank(),legend.position = "bottom")
# plot_start

##---------------------
#2. are all circular contigs recognized?
##---------------------

circ_analysis <- read_delim("Circularity_of_bacterial_contig.analysis", delim = "\t", escape_double = FALSE,  trim_ws = TRUE,skip = 1,col_names = c("SRA","contig","flye","Circlator","ReCycled"))%>% replace(is.na(.), "N") 
circ_analysis <- circ_analysis[!duplicated(circ_analysis),]

circ_analysis_long <- gather(circ_analysis, circulariation, outcome, c("Circlator","ReCycled"), factor_key=TRUE,na.rm = TRUE) %>% filter(!is.na(outcome))
circ_analysis_long$verdict <- ifelse(circ_analysis_long$flye==circ_analysis_long$outcome,"correct","incorrect")
totl_counts <- circ_analysis_long %>% group_by(circulariation) %>% summarise(total=n())

circ_analysis_sum <- circ_analysis_long %>% group_by(circulariation,verdict) %>% summarise(grouping=n()) %>% left_join(.,totl_counts,by="circulariation") %>% mutate(percent=100*(grouping/total))

circ_analysis_sum$verdict = factor(circ_analysis_sum$verdict, levels=c("incorrect","correct"))
countsGenomes <- circ_analysis_long %>% select(SRA,contig) %>% unique() %>% nrow()

colorss <- rev(c("#BBBE64","#8E5572"))

plot_circular <- ggplot(circ_analysis_sum,aes(x=circulariation,y=percent,fill=verdict,color=verdict))+geom_bar(stat = "identity")+theme_classic()+scale_fill_manual(values=colorss)+scale_color_manual(values=colorss)+
  labs(y="Genomes [%]",title = paste0("Correctly identified\ncircular contigs (n=",countsGenomes,")"))+
  theme(legend.title = element_blank(),axis.title.x = element_blank(),legend.position = "bottom")

# plot_circular
##---------------------
#3 speed
##---------------------

speed_analysis_01 <- read_delim("RunTime_Circlator.txt", delim = "\t", escape_double = FALSE,  trim_ws = TRUE,col_names = c("SRA","Circlator","ReCycled"))
speed_analysis_02 <- read_delim("RunTime_ReCycled.txt", delim = "\t", escape_double = FALSE,  trim_ws = TRUE,col_names = c("SRA","Circlator","ReCycled"))

speed_analysis <- rbind(speed_analysis_01,speed_analysis_02)
speed_analysis <- speed_analysis[!duplicated(speed_analysis),]
speed_analysis_long <- gather(speed_analysis, circulariation, outcome, c("Circlator","ReCycled"), factor_key=TRUE,na.rm = TRUE) %>% filter(!is.na(outcome)) 
speed_analysis_long_cleaned <- speed_analysis_long %>% group_by(SRA,circulariation) %>% summarise(outcome=median(outcome))%>% filter(outcome<50000)%>% filter(outcome>3)

countsGenomes <- speed_analysis_long$SRA %>% unique() %>% length()
 
library(ggquiver)
plot_speed <- ggplot(speed_analysis_long_cleaned,aes(x=circulariation,y=outcome,fill=circulariation ,color=circulariation ))+geom_violin(alpha=0.4)+ geom_jitter(shape=16, position=position_jitter(0.2))+theme_classic()+scale_y_continuous(trans = "log10",breaks=c(10,100,1000,10000))+  #+coord_trans(y="log10")+
  scale_fill_manual(values=colorss)+scale_color_manual(values=colorss)+
  labs(y="Runtime [seconds]",title = paste0("Runtime of the two tools\n(n=",countsGenomes,")"))+
  theme(legend.position = "none",axis.title.x = element_blank())
# plot_speed

speed_analysis_long_cleaned %>%  group_by(circulariation) %>% summarise(meanss=mean(outcome))

##---------------------
#4 together
##---------------------

library(patchwork)

plot_complete_genome+plot_circular+plot_start+plot_speed+ plot_layout(nrow=1)

# png("Plot_all.png", width = 4400, height = 3000,res=400)
pdf("Plot_all.pdf",width=11,height=7)

plot_complete_genome+plot_circular+plot_start+plot_speed+ plot_layout(nrow=1)
dev.off()
```

# check start alignement


Here, I reannotate the restarted contigs with (PROKKA)[https://github.com/tseemann/prokka] (Seemann, 2014) to check for proper restarting at dnaA. 

```{bash}




```

# make synteny non-start aligned and start-aligned

Here, I create the genome synteny plot with (Artemis)[https://www.sanger.ac.uk/tool/artemis-comparison-tool-act/] (Carver et al. 2005) and blast from genomes that are not restarted and the same genomes that are restarted with ReCycled. 

```{bash}
##--------------------------------------------------
#this analysis is done on the cluster
##--------------------------------------------------
    rm -r ${analysis_location}/Synteny/FASTA
    mkdir -p ${analysis_location}/Synteny/FASTA

      for sample in $(echo "24742 24740 24738 24736 24726")
        do
        
  echo "=============="
  echo ${sample}
  #echo "=============="

      cat   /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/*info*
            cat   /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/*.fasta > ${analysis_location}/Synteny/FASTA/${sample}.fasta
  
        done
    
##-----------------------
###bring local
##-----------------------  
mkdir -p /home/vincent/Desktop/Projects/2020_circleries/06_analysis/Synteny/FASTA
scp -r vincent@130.223.51.66:${analysis_location}/Synteny/FASTA/* /home/vincent/Desktop/Projects/2020_circleries/06_analysis/Synteny/FASTA

##-----------------------
###create a blastall crunch file
##-----------------------

cd /home/vincent/Desktop/Projects/2020_circleries/06_analysis/Synteny/FASTA

# run formatdb
formatdb -p F -i 24742.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24742.fasta -i 24740.fasta -o first.crunch -a 12

formatdb -p F -i 24740.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24740.fasta -i 24738.fasta -o second.crunch -a 12

formatdb -p F -i 24738.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24738.fasta -i 24736.fasta -o third.crunch -a 12

formatdb -p F -i 24736.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24736.fasta -i 24726.fasta -o fourth.crunch -a 12

conda activate myenv

##-----------------------
###artemis and act analysis
##-----------------------
act &

#select all genome and comparison files
#select SCORE CUTOFF > 1450
#select MINIMUM PERCENT CUTOFF >85

##-----------------------
###start-align genome
##-----------------------


mkdir -p ${analysis_location}/Synteny/FASTA_startAlign/

      for sample in $(echo "24742 24740 24738 24736 24726")
        do
        
  echo "=============="
  echo ${sample}
  #echo "=============="


   \
   #   --out-dir /data/Project/2020_StarterCultureDiversity/09_flye/assembly/${sample}/
     
/data/Project/2021_circlator/20220106_2020_circleries_tool/Circleries/Circleries.sh \
    -i ${analysis_location}/Synteny/FASTA/${sample}.fasta \
    -l /data/Project/2020_StarterCultureDiversity/01_Data/20200619_genomeSeqONT/01_rawdata/fastq/${sample}/${sample}_qualityFiltered_min7000_Q90.fastq.gz \
    -t 8 \
    -o  ${sample} \
    -d  ${analysis_location}/Synteny/FASTA_startAlign/ \
    -F -x

        done
  
  rm -r  ${analysis_location}/Synteny/FASTA_startAlign/tmp*

##-----------------------
###create a blastall crunch file
##-----------------------

cd /home/vincent/Desktop/Projects/2020_circleries/06_analysis/Synteny/FASTA_startAlign

# run formatdb
formatdb -p F -i 24742.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24742.fasta -i 24740.fasta -o first.crunch -a 12

formatdb -p F -i 24740.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24740.fasta -i 24738.fasta -o second.crunch -a 12

formatdb -p F -i 24738.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24738.fasta -i 24736.fasta -o third.crunch -a 12

formatdb -p F -i 24736.fasta
blastall -p blastn -m 8 -e 1e-5 -d 24736.fasta -i 24726.fasta -o fourth.crunch -a 12

conda activate myenv

##-----------------------
###artemis and act analysis
##-----------------------
act &
```

# References

- Eric Sayers, The E-utilities In-Depth: Parameters, Syntax and More. (2009)

- Hunt, M., Silva, N.D., Otto, T.D. et al. Circlator: automated circularization of genome assemblies using long sequencing reads. Genome Biol 16, 294 (2015). https://doi.org/10.1186/s13059-015-0849-0

- Kolmogorov, M., Yuan, J., Lin, Y. et al. Assembly of long, error-prone reads using repeat graphs. Nat Biotechnol 37, 540–546 (2019). https://doi.org/10.1038/s41587-019-0072-8

- Carver TJ, Rutherford KM, Berriman M, Rajandream MA, Barrell BG, Parkhill J. ACT: the Artemis Comparison Tool. Bioinformatics. 2005 Aug 15;21(16):3422-3. doi: 10.1093/bioinformatics/bti553. 

- Torsten Seemann, Prokka: rapid prokaryotic genome annotation, Bioinformatics, Volume 30, Issue 14, 15 July 2014, Pages 2068–2069, https://doi.org/10.1093/bioinformatics/btu153

- Somerville, V., Berthoud, H., Schmidt, R.S. et al. Functional strain redundancy and persistent phage infection in Swiss hard cheese starter cultures. ISME J 16, 388–399 (2022). https://doi.org/10.1038/s41396-021-01071-0
